tests/unit/phpDocumentor/AutoloaderLocatorTest.php

```diff
@@ -84,32 +84,72 @@ public function testAutoloadAtDefaultLocation() : void
         vfsStream::setup('root', null, $this->standaloneStructure);
         $baseDir = vfsStream::url('root/dummy/src/phpDocumentor');
         self::assertSame(
-            'vfs://root/dummy/src/phpDocumentor/../../vendor',
+            'vfs://root/dummy/vendor',
             AutoloaderLocator::findVendorPath($baseDir)
         );
     }
 
     public function testAutoloadComposerInstalled() : void
     {
         $root = vfsStream::setup('root', null, $this->composerInstalledStructure);
-        vfsStream::newFile('composer.json')->at($root->getChild('dummy'));
+        vfsStream::newFile('autoload.php')->at($root->getChild('dummy')->getChild('vendor'));
         $baseDir = vfsStream::url('root/dummy/vendor/phpDocumentor/phpDocumentor/src/phpDocumentor');
         $this->assertSame(
-            'vfs://root/dummy/vendor/phpDocumentor/phpDocumentor/src/phpDocumentor/../../../../../vendor',
+            'vfs://root/dummy/vendor',
             AutoloaderLocator::findVendorPath($baseDir)
         );
     }
 
     public function testAutoloadComposerInstalledCustomVendor() : void
     {
         $root = vfsStream::setup('root', null, $this->customVendorDir);
-        vfsStream::newFile('composer.json')
-            ->withContent($this->customVendorDirComposer)
-            ->at($root->getChild('dummy'));
+        vfsStream::newFile('autoload.php')->at($root->getChild('dummy')->getChild('custom-vendor'));
         $baseDir = vfsStream::url('root/dummy/custom-vendor/phpDocumentor/phpDocumentor/src/phpDocumentor');
         $this->assertSame(
-            'vfs://root/dummy/custom-vendor/phpDocumentor/phpDocumentor/src/phpDocumentor/../../../../../custom-vendor',
+            'vfs://root/dummy/custom-vendor',
             AutoloaderLocator::findVendorPath($baseDir)
         );
     }
+
+    public function testAutoloadComposerNotFindableVendor() : void
+    {
+        $root = vfsStream::setup('root', null, []);
+        $baseDir = vfsStream::url('root/dummy/custom-vendor/phpDocumentor/phpDocumentor/src/phpDocumentor');
+        $this->expectException(\RuntimeException::class);
+        $this->expectExceptionMessage('Unable to find vendor directory for vfs://root/dummy/custom-vendor/phpDocumentor/phpDocumentor/src/phpDocumentor');
+        AutoloaderLocator::findVendorPath($baseDir);
+    }
+}
+
+/**
+ * This function overrides the native realpath($url) function, removing
+ * all the "..", ".", "///" of an url. Contrary to the native one, 
+ * 
+ * @see https://github.com/bovigo/vfsStream/issues/207
+ * @param string $url
+ * @param string|bool The cleaned url or false if it doesn't exist
+ */
+function realpath(string $url)
```

Comment:
This is unnecessary since realpath should not be used; the vfs filesystem has the same issues as the phar filesystem; with realpath you break out of virtual filesystem